packages/app/src/app/components/CreateNewSandbox/CreateSandbox/Import/Import.tsx

```diff
@@ -1,174 +1,142 @@
-          style={{ display: 'inline' }}
-          href="/docs/importing#import-from-github"
-        >
-          <StyledInfoIcon />
-        </IconLink>
-        <small>
-          {error ? (
-            <PlaceHolderLink>{error}</PlaceHolderLink>
-          ) : (
-            'Tip: CodeSandbox Projects is a new experience with built-in source control and support for VSCode, perfect for building projects of any size.'
-          )}
-        </small>
-      </FeatureText>
-    </form>
+    </Stack>
   );
 };
 
-export const Import = () => {
-  useEffect(() => {
-    track('Create Sandbox Tab Open', { tab: 'import' });
-  }, []);
+type UrlState = {
+  raw: string;
+  parsed: { owner: string; name: string } | null;
+  error: string | null;
+};
 
-  return (
-    <>
-      <Header>
-        <span>Import from GitHub</span>
-      </Header>
+type ImportProps = {
+  onRepoSelect: (repo: GithubRepoToImport) => void;
+};
+export const Import: React.FC<ImportProps> = ({ onRepoSelect }) => {
+  const { hasLogIn } = useAppState();
 
-      <Features>
-        <GitHubIcon css={{ width: 69, height: 69 }} />
+  const [url, setUrl] = React.useState<UrlState>({
+    raw: '',
+    parsed: null,
+    error: null,
+  });
+  const [shouldFetch, setShouldFetch] = React.useState(false);
+  const githubRepo = useGithubRepo({
+    owner: url.parsed?.owner,
+    name: url.parsed?.name,
+    shouldFetch,
+    onCompleted: onRepoSelect,
+  });
 
-        <FeatureText css={{ marginTop: '25px' }}>
-          Enter the URL to your GitHub repository to import
-          <IconLink
-            style={{ display: 'inline' }}
-            href="/docs/projects/learn/introduction/about-projects"
-          >
-            <StyledInfoIcon />
-          </IconLink>
-        </FeatureText>
+  const handleUrlInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    if (!value) {
+      setUrl({ raw: value, parsed: null, error: null });
+    } else if (!gitHubRepoPattern.test(value)) {
+      setUrl({
+        raw: value,
+        parsed: null,
+        error: 'The URL provided is not valid.',
+      });
+    } else {
+      const { owner, name } = getOwnerAndNameFromInput(value.trim());
+      setUrl({
+        raw: value,
+        parsed: {
+          owner,
+          name,
+        },
+        error: null,
+      });
+    }
+  };
 
-        <ImportFromGithub />
-      </Features>
+  const handleFormSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setShouldFetch(true);
+  };
 
-      <ImportChoices>
-        <a href="/docs/importing#export-with-cli">
-          <TerminalIcon />
-          CLI Documentation
-        </a>
-        <a href="/docs/importing#define-api">
-          <DownloadIcon />
-          API Documentation
-        </a>
-      </ImportChoices>
-    </>
+  if (!hasLogIn) {
+    return <UnauthenticatedImport />;
+  }
+
+  return (
+    <Stack direction="vertical" gap={4}>
+      <Text
+        as="h2"
```

Comment:
should this be a heading element? feels more like a label for the input

Reply:
@alexnm that's an excellent question. Because of how it looks, I did set it as a heading but I'm using its contents to [describe](https://github.com/codesandbox/codesandbox-client/pull/6957/files/7931f461fb7a8fe447409170395b0e773a974d6e#diff-837cda56745dc61fcef9cb54609c620e8d20380214937cde270579e1da1ab623R111) the input element for assistive technologies.